state/executor.go

```diff
@@ -218,31 +218,40 @@ func (e *Executor) BeginTxn(
 		PostHook:    e.PostHook,
 	}
 
-	// enable contract deployment allow list (if any)
-	if e.config.ContractDeployerAllowList != nil {
-		txn.deploymentAllowList = addresslist.NewAddressList(txn, contracts.AllowListContractsAddr)
+	// contract deployment access list should be enabled
+	// either if access lists super admin is defined or access lists are defined in the genesis
```

Comment:
So, all AccessLists will be enabled if SuperAdmin is defined? This should be stated in the docs at it has an impact in Gas costs for each transaction

Reply:
Yes, because super admin can enable/disable list at any time. By default if no role is specified, list will be disabled and there will be no role checking on transaction execution/smart contract deploy.

Yes, we should probably update the docs.  I have used the same predefined read/write gas costs for these new functions:
- `function setListEnabled(bool)`
- `function getListEnabled() returns (bool)`

Reply:
I just realized, what was your first question. No, access list will be enabled if `getListEnabled` value in storage is set to true. If just superadmin is specified, then initial value will be `setListEnabled(false)`. But we need to create list on order to allow superadmin to change enabled/disabled value